src/Main.cpp

```diff
@@ -152,19 +208,73 @@ struct PyramidPlunder : public Game
 
 	void on_start() override
 	{
-		auto player = spawn()
-			.with<Player>(SpecialBlindfold::HumanEyes)
-			.with<Sprite>(ecs::no_entity)
-			.with<SpriteAnimation>(Spritesheet::get_by_name("pyramidplunder/archaeologist_standing"))
-			.with<Visibility>(true)
-			.with<Position>(geometry::Vec2{ 300, 100 })
-			.with<Movement>(2000.0f, 50.0f)
-			.with<KeyBinding>(KeyCode::KEY_LEFT, KeyCode::KEY_DOWN, KeyCode::KEY_UP, KeyCode::KEY_RIGHT, KeyCode::KEY_SPACE)
-			.done();
+		level_manager = new LevelManager();
+		level_manager->load_level("Levels/level1.txt");
+
+		for(int i = 0; i < TILE_ROWS; i++)
+		{
+        	for(int j = 0; j < TILE_COLS; j++)
+			{
+				Char c = level_manager->level_map[j][i];
+				if(c == 'x')
+				{
+						spawn()
+						.with<Sprite>(ecs::no_entity, 6)
+						.with<SpriteAnimation>(Spritesheet::get_by_name("pyramidplunder/wall"))
+						.with<Visibility>(true)
+						.with<Position>(geometry::Vec2{ i*96 + 25, j*96 + 25})
+						.done();
+				}
+				if(c == 'o' || c == 'a' || c == 'b')
+				{
+						spawn()
+						.with<Sprite>(ecs::no_entity, 2)
+						.with<SpriteAnimation>(Spritesheet::get_by_name("pyramidplunder/sand"))
+						.with<Visibility>(true)
+						.with<Position>(geometry::Vec2{ i*96 +25, j*96+25})
+						.done();
+				}
+				if(c == 'b')
+				{
+						spawn()
+						.with<Sprite>(ecs::no_entity, 5)
+						.with<SpriteAnimation>(Spritesheet::get_by_name("box/box_1"))
+						.with<Visibility>(true)
+						.with<Position>(geometry::Vec2{ i*96, j*96})
+						.with<Movement>(2000.0f, 50.0f, 32, 32)
+						.done();
+				}
+				if(c == 'p')
+				{
+						spawn()
+						.with<Sprite>(ecs::no_entity, 4)
+						.with<SpriteAnimation>(Spritesheet::get_by_name("pushplate/pushplate_3"))
+						.with<Visibility>(true)
+						.with<Position>(geometry::Vec2{ i*96, j*96})
+						.done();
+				}
+				if(c == 'a')
+				{
+						spawn()
+						.with<Player>(SpecialBlindfold::HumanEyes)
+						.with<Sprite>(ecs::no_entity, 6)
+						.with<SpriteAnimation>(Spritesheet::get_by_name("pyramidplunder/archaeologist_standing"))
+						.with<Visibility>(true)
+						.with<Position>(geometry::Vec2{ i*96, j*96})
+						.with<Movement>(3000.0f, 50.0f, 16, 16)
+						.with<KeyBinding>(KeyCode::KEY_LEFT, KeyCode::KEY_DOWN, KeyCode::KEY_UP, KeyCode::KEY_RIGHT, KeyCode::KEY_SPACE)
+						.done();
+				}
+        	}
+		}
+	}
+
+	void on_end() override
+	{
+		delete level_manager;
```

Comment:
This would be automated if it wasn't a raw_ptr... Btw, why is it a raw pointer? If you actually want to do a pointer, make it a unique_ptr instead as you're not sharing it anywhere else, or a shared_ptr if you are! Discuss below!

Reply:
Honestly, don't know why we used raw pointer here and don't think it's necessary. I'm not that much familiar with smart pointers and their semantics, so I'm just going to remove pointer completely for now. :)